packages/py-api/py_api/functionality/hackathon/teams/teams_utility_functions.py

```diff
@@ -0,0 +1,117 @@
+from typing import Any, Dict, List, Optional
+
+from bson import ObjectId
+from py_api.database.initialize import t_col
+from py_api.models import HackathonTeam
+from py_api.models.hackathon_teams_models import TeamType
+from pymongo.results import UpdateResult
+
+
+class TeamsUtilities:
+
+    @classmethod
+    def create_team(
+            cls, user_id: str,
+            team_name: str, ) -> HackathonTeam | None:
+        team_type = TeamType.NORMAL
+
+        if not team_name:
+            # If no team_name is provided during registration, the participant is registering individually.
+            # We create a team of type RANDOM which should be filled with such participants
+            team_type = TeamType.RANDOM
+            team_name = cls.generate_random_team_name()
+
+        if cls.fetch_team(team_name=team_name):
+            # A name with this team_name exist so no new team is created
+            return None
+
+        # The team is created with one person, and they are the admin of it
+        team_members_ids = [user_id]
+
+        new_team = HackathonTeam(
+            team_name=team_name,
+            team_type=team_type, team_members=team_members_ids,
+        )
+        t_col.insert_one(new_team.model_dump())
+
+        return new_team
+
+    @classmethod
+    def add_participant_to_team(
+            cls, team_name: str,
+            user_id: str,
+    ) -> HackathonTeam | None | str:
+        team = cls.fetch_team(team_name=team_name)
+
+        if not team:
+            return None
+
+        if len(team.team_members) == 6:
+            return "No space left"  # TODO: This should be fixed
+
+        team.team_members.append(user_id)
+
+        return team
+
+    @classmethod
+    def fetch_team(
+            cls, team_name: str = "",
+            team_id: str = "",
+    ) -> HackathonTeam | None:
+        """Fetches team by team_name or team_id"""
+
```

Comment:
Add TODO comment -> use OR query for filtering similarly to how it's done in `update_team_query`